segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ContainerRecoveryUtils.java

```diff
@@ -307,8 +307,9 @@ public static void updateCoreAttributes(Map<Integer, String> backUpMetadataSegme
                                             ExecutorService executorService,
                                             Duration timeout) throws InterruptedException, ExecutionException,
             TimeoutException {
-        Preconditions.checkState(backUpMetadataSegments.size() == containersMap.size(), "The number of " +
-                "back-up metadata segments and containers should match.");
+        Preconditions.checkNotNull(backUpMetadataSegments, "Back up metadata segments");
+        Preconditions.checkState(backUpMetadataSegments.size() <= containersMap.size(), "The number of " +
+                "back-up metadata segments should be less or equal to the number of debug segment containers.");
```

Comment:
sometimes not all container metadata segments exist in the storage, so their back ups are not created. So, "==" check failed. 
In case `backUpMetadataSegments` is empty, this would still pass, and update attributes will happen.